lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -492,33 +508,44 @@ def _convert_bbox_coordinates(
         y2 = height - y2
         return x1, y1, x2, y2
 
-    def extract(self, filename: Union[str, IOBase]) -> List[List[Element]]:
+    def extract(self, filename: Union[str, IOBase], cache_results=True, cache_location="") -> List[List[Element]]:
         # The naming is slightly confusing, but `open_filename` accepts either
         # a filename (str) or a file-like object (IOBase)
-        with open_filename(filename, "rb") as fp:
-            fp = cast(BinaryIO, fp)
-            pages = []
-            for page, page_layout in self._open_pdfminer_pages_generator(fp):
-                width = page_layout.width
-                height = page_layout.height
-                texts: List[Element] = []
-                for obj in page_layout:
-                    x1, y1, x2, y2 = self._convert_bbox_coordinates(obj.bbox, height)
-
-                    if hasattr(obj, "get_text"):
-                        text = Element()
-                        text.type = "text"
-                        text.bbox = BoundingBox(x1 / width, y1 / height, x2 / width, y2 / height)
-                        text.text_representation = obj.get_text()
-                        if text.text_representation:
-                            texts.append(text)
-
-                pages.append(texts)
-            return pages
+        cache_dir = cache_location if cache_location else os.path.dirname(__file__)
+        cache_loc = cache_dir + "/PDFMiner_Cache"
```

Comment:
Shouldn't this be done _once_ when you initialize sycamore?  It seems a bit awkward that you specify the cache location on every invocation of the method (although I assume in most cases, the default will be chosen).  Actually, maybe we shouldn't allow the caller to be able to specify/change the cache location at all.